common/system.h

```diff
@@ -1147,238 +1243,264 @@ class OSystem : Common::NonCopyable {
-	 * recursive locking. That is, a thread may lock a mutex twice w/o
-	 * deadlocking. In case of a multilock, the mutex has to be unlocked
+	 * recursive locking. That is, a thread can lock a mutex twice without
+	 * deadlocking. In case of a multilock, the mutex must be unlocked
 	 * as many times as it was locked befored it really becomes unlocked.
 	 *
-	 * @param mutex	the mutex to lock.
+	 * @param mutex	The mutex to lock.
 	 */
 	virtual void lockMutex(MutexRef mutex) = 0;
 
 	/**
 	 * Unlock the given mutex.
-	 * @param mutex	the mutex to unlock.
+	 *
+	 * @param mutex	The mutex to unlock.
 	 */
 	virtual void unlockMutex(MutexRef mutex) = 0;
 
 	/**
-	 * Delete the given mutex. Make sure the mutex is unlocked before you delete it.
-	 * If you delete a locked mutex, the behavior is undefined, in particular, your
-	 * program may crash.
-	 * @param mutex	the mutex to delete.
+	 * Delete the given mutex.
+	 *
+	 * Make sure the mutex is unlocked before you delete it.
+	 * If you delete a locked mutex, the behavior is undefined.
+	 * In particular, your program may crash.
+	 *
+	 * @param mutex	The mutex to delete.
 	 */
 	virtual void deleteMutex(MutexRef mutex) = 0;
 
-	//@}
+	/** @} */
 
 
 
-	/** @name Sound */
-	//@{
+	/** @defgroup common_system_sound Sound
+	 *  @ingroup common_system
+	 *  @{
+	 */
 
 	/**
-	 * Return the audio mixer. For more information, refer to the
-	 * Audio::Mixer documentation.
+	 * Return the audio mixer.
+	 *
+	 * For more information, see @ref Audio::Mixer.
 	 */
 	virtual Audio::Mixer *getMixer() = 0;
 
-	//@}
+	/** @} */
 
 
 
-	/** @name Audio CD */
-	//@{
+	/** @defgroup common_system_audio Audio CD
+	 *  @ingroup common_system
+	 *  @{
+	 */
 
 	/**
-	 * Return the audio cd manager. For more information, refer to the
-	 * AudioCDManager documentation.
+	 * Return the audio CD manager.
+	 *
+	 * For more information, see @ref AudioCDManager.
 	 */
 	inline AudioCDManager *getAudioCDManager() {
 		return _audiocdManager;
 	}
 
-	//@}
+	/** @} */
 
 
 
-	/** @name Miscellaneous */
-	//@{
+	/** @defgroup common_system_misc Miscellaneous
+	 *  @ingroup common_system
+	 *  @{
+	 */
+
 	/** Quit (exit) the application. */
 	virtual void quit() = 0;
 
 	/**
-	 * Signals that a fatal error inside the client code has happened.
+	 * Signal that a fatal error inside the client code has occurred.
 	 *
 	 * This should quit the application.
 	 */
 	virtual void fatalError();
 
 	/**
 	 * Set a window caption or any other comparable status display to the
-	 * given value. The caption must be a pure ISO LATIN 1 string. Passing a
-	 * string with a different encoding may lead to unexpected behavior,
+	 * given value.
+	 *
+	 * The caption must be a pure ISO LATIN 1 string. Passing a string
+	 * with a different encoding may lead to unexpected behavior,
 	 * even crashes.
 	 *
-	 * @param caption	the window caption to use, as an ISO LATIN 1 string
+	 * @param caption The window caption to use, as an ISO LATIN 1 string.
```

Comment:
As a warning, there is currently another pull request open (PR #2588) to use `Common::U32String` instead of `const char*`. We will need to revisit that doc when it is merged (since then the `ISO LATIN 1` will no longer be correct as it will use UTF-32 encoding instead).